back/app/services/invalid_data_checker.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+class InvalidDataChecker
+  SKIP_CLASSES = %w[
+    EmailCampaigns::Delivery
+    CommonPassword
+    Analytics::FactVisit
+  ].freeze
+
+  def check_global(summary: nil)
+    summary ||= initial_summary
+    Cl2DataListingService.new.cl2_global_models.each do |claz|
+      claz.all.find_each do |object|
+        errors = validation_errors object
+        add_errors_to_summary errors, claz.name, object.id, summary if errors
+      end
+    end
+
+    summary
+  end
+
+  def check_tenant(tenant: nil, summary: nil)
+    tenant ||= Tenant.current
+    summary ||= initial_summary
+    Apartment::Tenant.switch(tenant.schema_name) do
```

Comment:
Wonder if the tenant switching makes more sense to stay in the rake task so the service always just uses the current tenant. Might make it more reusable elsewhere (if we ever needed to)

Reply:
I'm not sure how switching inside `check_tenant` makes it less reusable? It's possible to either call this method while passing a tenant, or you could still do a switch around it (in which case `check_tenant` will use `Tenant.current`).